internal/backends/python/python.go

```diff
@@ -769,20 +762,84 @@ func makePythonUvBackend() api.LanguageBackend {
 
 		return pkgs
 	}
-	addExtraIndexes := func(pkgName string) {
-		extraIndexes, ok := extraIndexMap[pkgName]
-		if ok {
-			uvIndex := os.Getenv("UV_INDEX")
-
-			for _, index := range extraIndexes {
-				if strings.HasPrefix(runtime.GOOS, index.os) {
-					uvIndex = index.url + " " + uvIndex
+	addTorchOverride := func() error {
+		if !util.TomlEditorIsAvailable() {
+			return errors.New(
+				"toml-editor is not on the PATH, please install it with " +
+					"`nix profile install github:replit/toml-editor` or " +
+					"`cargo install --git https://github.com/replit/toml-editor` and ensure it's on the PATH.",
+			)
+		}
+
+		// check if the torch cpu index is already present
+		getOp := util.TomlEditorOp{
+			Op:   "get",
+			Path: "tool/uv/index",
+		}
+		response, err := util.ExecTomlEditor("pyproject.toml", []util.TomlEditorOp{getOp})
+		if err != nil {
+			return err
+		}
+		if len(response.Results) != 1 {
+			return fmt.Errorf("expected one result")
+		}
+		result := response.Results[0]
+		if result != nil {
+			if arr, ok := result.([]interface{}); ok {
+				if slices.ContainsFunc(arr, func(val interface{}) bool {
+					gotValue, ok := val.(map[string]interface{})
+					if !ok {
+						return false
+					}
+
+					url, ok := gotValue["url"]
+					if !ok {
+						return false
+					}
+
+					return url == "https://download.pytorch.org/whl/cpu"
+				}) {
+					return nil
 				}
```

Comment:
As much as it pains me to say this, the `ContainsFunc` is actually harder to read. I'd suggest:

```suggestion
				for value, _ := range arr {
					index, ok := value.(map[string]interface{})
					if !ok {
						continue
					}

					name, ok := index["name"]
					if !ok {
						continue
					}

					hasIndex := name == "pytorch-cpu"
					break
				}
```